keyvault/client/src/main/java/com/azure/keyvault/SecretService.java

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault;
+
+import com.azure.common.annotations.ExpectedResponses;
+import com.azure.common.annotations.Host;
+import com.azure.common.annotations.PUT;
+import com.azure.common.annotations.UnexpectedResponseExceptionType;
+import com.azure.common.annotations.HostParam;
+import com.azure.common.annotations.QueryParam;
+import com.azure.common.annotations.HeaderParam;
+import com.azure.common.annotations.BodyParam;
+import com.azure.common.annotations.PathParam;
+import com.azure.common.annotations.POST;
+import com.azure.common.annotations.PATCH;
+import com.azure.common.annotations.DELETE;
+import com.azure.common.annotations.GET;
+import com.azure.common.http.rest.RestException;
+import com.azure.common.http.rest.RestResponse;
+import com.azure.keyvault.implementation.Page;
+import com.azure.keyvault.models.DeletedSecret;
+import com.azure.keyvault.models.Secret;
+import com.azure.keyvault.models.SecretBackup;
+import com.azure.keyvault.models.SecretRestoreRequestOptions;
+import com.azure.keyvault.models.SecretRequestOptions;
+import com.azure.keyvault.models.SecretInfo;
+import reactor.core.publisher.Mono;
+
+/**
+ * The interface defining all the services for {@link SecretClient} to be used
+ * by the proxy service to perform REST calls.
+ *
+ * This is package-private so that these REST calls are transparent to the user.
+ */
+@Host("{url}")
+interface SecretService {
+
+    @PUT("secrets/{secret-name}")
+    @ExpectedResponses({200})
+    @UnexpectedResponseExceptionType(RestException.class)
```

Comment:
When our exception hierarchy is implemented (see #3229), we should throw richer exception types here //cc @conniey - however, I'm not sure how `UnexpectedResponseExceptionType` will support that - @anuchandy - do you think we need to improve `UnexpectedResponseExceptionType` such that we can throw different exception types based on the error code returned?

Reply:
I have an issue for that #3139.

From how I saw UnexpectedResponseExceptionType used in code is that it has to extend from RestException. And if you set it to lets say, ForbiddenException that has a `String body()` method, then it would try to deserialize an exception into that `String` and create that exception class with that body. `RestProxy.instantiateUnexpectedException` is where most of the stuff I saw is.

Reply:
My primary question to resolve is: shouldn't we be able to state in this annotation (or multiple, repeatable instances of this annotation) that the exception type will be XException for status code X, and YException for status code Y, otherwise default to ZException? // @anuchandy 

E.g. would something like this make sense:
```java
@PUT("secrets/{secret-name}")
@ExpectedResponses({200})
@UnexpectedResponseExceptionType(statusCode = 333, type = Exception333.class) // for the status code '333' case only
@UnexpectedResponseExceptionType(statusCode = {345, 346}, type = RestException.class) // for status code 345 and 346 only will we throw RestException
@UnexpectedResponseExceptionType(ServiceRequestException.class) // default case, for all other status codes - will be this exception
```

Reply:
Ahh, that makes sense. I see that pattern used in a few swagger clients.

Reply:
JonathanGiles agree, we can improve this, duplicate annotations are not allowed so we need to do something like:


```java
@UnexpectedResponseExceptionTypes({
    @UnexpectedResponseExceptionTypes.Type(code = {333}, value = Exception333.class),
    @UnexpectedResponseExceptionTypes.Type(code = {345, 346}, value = RestException.class),
    @UnexpectedResponseExceptionTypes.Type(value = ServiceRequestException.class)
})
```

Where the annotation looks like:

```java
public @interface UnexpectedResponseExceptionTypes {
    public Type[] value();

    public @interface Type {
        public int[] code() default { };
        public Class<?> value();
    }
}
```

Reply:
@JonathanGiles @conniey created an issue to track this https://github.com/Azure/azure-sdk-for-java/issues/3302